core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -1992,8 +1995,16 @@ wasmtime_ssp_path_open(wasm_exec_env_t exec_env, struct fd_table *curfds,
         noflags |= O_SYNC;
         needed_inheriting |= __WASI_RIGHT_FD_SYNC;
     }
-    if (write && (noflags & (O_APPEND | O_TRUNC)) == 0)
-        needed_inheriting |= __WASI_RIGHT_FD_SEEK;
+#else
+    // It's not clear whether we want to support these flags in the standard
+    // yet: https://github.com/WebAssembly/wasi-filesystem/issues/98 so disable
```

Comment:
I don't think this comment is related. It's about preview 2, whereas this function is for preview 1 only. And those flags are already part of the preview 1 standard: https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/docs.md#record-members-4
My suggestion is to update the preview1 doc saying that this feature is not available on some platforms therefore `path_open` might return `ENOTSUP` in that case.

Reply:
Do you agree with the suggestion that yamt proposed above?
https://github.com/bytecodealliance/wasm-micro-runtime/pull/2581#discussion_r1341270520

Reply:
yes, I think that makes sense.

Reply:
Removed the comment, thanks. I'll raise a PR shortly on the docs to clarify that sync behaviour is platform-dependent.

Reply:
https://github.com/WebAssembly/WASI/pull/560